### PR TITLE
Enabled support for a token issuance endpoint in Alfred

### PR DESCRIPTION
**Pull request by dhr-verma**
## Description

This PR adds support for a fluid token issuance endpoint. This endpoint can be used to issue fluid access tokens based on a custom implementation of the `IFluidAccessTokenGenerator` interface. One use of this endpoint is to enable access using cloud identity providers such as Entra-ID. Alfred has a new endpoint - `api/v1/tenants/:tenantid/accesstoken`. This endpoint expects a `Bearer` token and creates an access token for a given fluid tenant.

This PR also adds support to inject a custom implementation of the `IFluidAccessTokenGenerator`. This can be used to implement any custom logic that is needed for a business use case.

It also has unit tests for the following:

1) Throttling of the endpoint
2) Validation of cases where a `Bearer` token is not provided or an invalid authorization method is used
3) Validation of a valid token creation path
4) Validation of token creation failure due to invalid token signature and due to unauthorized access

## Breaking Changes

This PR adds a new resourceFactory arg of type `IFluidAccessTokenGenerator`. This arg is customizable as well.

### CODE REVIEW

File: server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/api.ts [hunk 1]
```diff
@@ -49,6 +49,7 @@ export function create(
 	tenantManager: core.ITenantManager,
 	storage: core.IDocumentStorage,
 	tenantThrottlers: Map<string, core.IThrottler>,
+	fluidAccessTokenGenerator: core.IFluidAccessTokenGenerator,
```
Comment (yunho-microsoft) [inline server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/api.ts:52]: Wonder if we want to make it optional in OSS?

Comment (dhr-verma) [reply server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/api.ts:52]: Yeah that's a good idea. I can create an endpoint only if this implementation is injected. However, I would still like to keep the tests as they test API calls from `Alfred` -> `A token generation API`.

**Review by yunho-microsoft — APPROVED** [review]

File: server/routerlicious/.changeset/chubby-worms-wonder.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+"@fluidframework/server-services-core": minor
+---
+
+Added a new interface - IFluidAccessTokenGenerator
+
+This interface can be implemented to create an access token for the service
```
Comment (tylerbutler) [inline server/routerlicious/.changeset/chubby-worms-wonder.md:7]:
```suggestion
The `IFluidAccessTokenGenerator` interface can be implemented to create an access token for the service.
```

File: server/routerlicious/.changeset/real-eels-accept.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+"@fluidframework/server-routerlicious-base": minor
+---
+
+Adds a new resource type - IFluidAccessTokenGenerator
+
+This resource type can be used by a new endpoint `/api/v1/tenants/:tenantId/accesstoken` to generate an access token for the service. This resource is customizable and can be injected into into the endpoint.
```
Comment (tylerbutler) [inline server/routerlicious/.changeset/real-eels-accept.md:7]:
```suggestion
This resource type can be used by a new endpoint, `/api/v1/tenants/:tenantId/accesstoken`, to generate an access token for the service. This resource is customizable and can be injected into the endpoint.
```

**Review by tylerbutler — APPROVED** [review]
Couple of nits in the docs.

**Pull request merged**